spec.html

```diff
@@ -9057,9 +9184,12 @@ <h1>
       <emu-alg>
         1. Return InstantiateAsyncArrowFunctionExpression of |AsyncArrowFunction| with argument _name_.
       </emu-alg>
-      <emu-grammar>ClassExpression : `class` ClassTail</emu-grammar>
+      <emu-grammar>ClassExpression : DecoratorList? `class` ClassTail</emu-grammar>
       <emu-alg>
-        1. Let _value_ be ? ClassDefinitionEvaluation of |ClassTail| with arguments *undefined* and _name_.
+        1. If |DecoratorList?| is present, then
+          1. Let _decorators_ be ? DecoratorListEvaluation of |DecoratorList|.
+        1. Else, let _decorators_ be a new empty List.
+        1. Let _value_ be ? ClassDefinitionEvaluation of |ClassTail| with arguments *undefined*, _name_, and _decorators_.
```

Comment:
ClassDefinitionEvaluation requires its 3rd arg to be a List of function objects, but I don't think there's anything here to ensure that `_decorators_` is such a List. Similarly for some other invocations of ClassDefinitionEvaluation.

(Maybe ClassDefinitionEvalution should accept a List of ES language values, and then deal with the possibility that one or more might not be a function object.)

(Similarly, BindingClassDeclarationEvaluation should maybe accept a List of ES language values.)

Reply:
The list is actually of `DecoratorDefinition` records since decorators need to have their receiver and the value. I've updated the `DecoratorDefinition` record to store any ECMAScript language value, and when that value is passed to `Call` during application that will throw a type error if it is not callable.